Documentation/building/linux-instructions.md

```diff
@@ -99,36 +102,57 @@ D:\git\coreclr> build.cmd linuxmscorlib
 
-Build the Framework Native Components
-======================================
+Build the Framework
+===================
 
 ```
-ellismg@linux:~/git/corefx$ src/Native/build.sh
-ellismg@linux:~/git/corefx$ cp bin/Linux.x64.Debug/Native/*.so ~/coreclr-demo/runtime
+ellismg@linux:~/git/corefx$ ./build.sh
 ```
 
-Build the Framework Managed Components
-======================================
+For the purposes of Hello World, you need to copy a few required files to the demo folder.
 
-We don't _yet_ have support for building managed code on Linux, so you'll need a Windows machine with clones of CoreFX project.
+```
+ellismg@linux:~/git/corefx$ cp bin/Linux.x64.Debug/Native/*.so ~/coreclr-demo/runtime
+ellismg@linux:~/git/corefx$ cp bin/Linux.AnyCPU.Debug/System.Console/System.Console.dll ~/coreclr-demo/runtime
+ellismg@linux:~/git/corefx$ cp bin/Linux.AnyCPU.Debug/System.Diagnostics.Debug/System.Diagnostics.Debug.dll ~/coreclr-demo/runtime
+```
 
-You will build the rest of the framework that out of the corefx repository.  
-You need to pass some special parameters to build.cmd when building out of the CoreFX repository.
+The runtime directory should now look like the following:
 
 ```
-D:\git\corefx> build.cmd /p:OSGroup=Linux /p:SkipTests=true
+matell@linux:~$ ls ~/coreclr-demo/runtime/
+corerun             System.Diagnostics.Debug.dll
+libcoreclr.so       System.Native.so
+mscorlib.dll        System.Net.Http.Native.so
+System.Console.dll  System.Security.Cryptography.Native.so
 ```
 
-It's also possible to add `/t:rebuild` to the build.cmd to force it to delete the previously built assemblies.
+(Alternative) Build the Framework Managed Components on Windows
+===============================================================
 
-For the purposes of Hello World, you need to copy over both `bin\Linux.AnyCPU.Debug\System.Console\System.Console.dll` and `bin\Linux.AnyCPU.Debug\System.Diagnostics.Debug\System.Diagnostics.Debug.dll`  into the runtime folder on Linux. (e.g `~/coreclr-demo/runtime`).
+If the build fails (for example due to Mono issues), alternatively you can build it on Windows.
+You'll need a Windows machine with a clone of the CoreFX repository.
 
-After you've done these steps, the runtime directory on Linux should look like this:
+Build the Native Components on Linux
+------------------------------------
 
 ```
-matell@linux:~$ ls ~/coreclr-demo/runtime/
-corerun  libcoreclr.so  mscorlib.dll  System.Console.dll  System.Diagnostics.Debug.dll System.Native.so
+ellismg@linux:~/git/corefx$ src/Native/build.sh
+ellismg@linux:~/git/corefx$ cp bin/Linux.x64.Debug/Native/*.so ~/coreclr-demo/runtime
+```
+
+Build the Managed Components on Windows
+---------------------------------------
+
+You need to pass some special parameters to build.cmd when building out of the CoreFx repository for Linux.
+
 ```
+D:\git\corefx> build.cmd /p:OSGroup=Linux /p:SkipTests=true
+```
```

Comment:
I put instructions for building CoreFx on Linux with build.sh as the primary way to build the framework.  I thought it would be useful to keep instructions for how to build CoreFx on Windows as an alternative in case the build on Windows fails, just like the instructions have the alternative for building mscorlib on Windows in case the Linux build of CoreClr fails.

The Mac instructions don't have the alternative listed for building CoreFx, so maybe it's not really necessary.  You probably build on Mono a lot more than I do, so would you say that it's not useful to have the alternative instructions for building CoreFx on Windows at this point?